crates/bevy_ecs/src/schedule_v3/executor/single_threaded.rs

```diff
@@ -0,0 +1,141 @@
+    fn kind(&self) -> ExecutorKind {
+        ExecutorKind::SingleThreaded
+    }
+
+    fn init(&mut self, schedule: &SystemSchedule) {
+        // pre-allocate space
+        let sys_count = schedule.system_ids.len();
+        let set_count = schedule.set_ids.len();
+        self.evaluated_sets = FixedBitSet::with_capacity(set_count);
+        self.completed_systems = FixedBitSet::with_capacity(sys_count);
+        self.unapplied_systems = FixedBitSet::with_capacity(sys_count);
+    }
+
+    fn run(&mut self, schedule: &mut SystemSchedule, world: &mut World) {
+        for system_index in 0..schedule.systems.len() {
+            #[cfg(feature = "trace")]
+            let name = schedule.systems[system_index].get_mut().name();
+            #[cfg(feature = "trace")]
+            let should_run_span = info_span!("check_conditions", name = &*name).entered();
+
+            let mut should_run = !self.completed_systems.contains(system_index);
+            for set_idx in schedule.sets_of_systems[system_index].ones() {
+                if self.evaluated_sets.contains(set_idx) {
+                    continue;
+                }
+
+                // evaluate system set's conditions
+                let set_conditions_met = evaluate_and_fold_conditions(
+                    schedule.set_conditions[set_idx].get_mut().as_mut(),
+                    world,
+                );
+
+                if !set_conditions_met {
+                    self.completed_systems
+                        .union_with(&schedule.systems_in_sets[set_idx]);
+                }
+
+                should_run &= set_conditions_met;
+                self.evaluated_sets.insert(set_idx);
+            }
+
+            // evaluate system's conditions
+            let system_conditions_met = evaluate_and_fold_conditions(
+                schedule.system_conditions[system_index].get_mut().as_mut(),
+                world,
+            );
+
+            should_run &= system_conditions_met;
+
+            #[cfg(feature = "trace")]
+            should_run_span.exit();
+
+            // system has either been skipped or will run
+            self.completed_systems.insert(system_index);
+
+            if !should_run {
+                continue;
+            }
+
+            let system = schedule.systems[system_index].get_mut();
+            if is_apply_system_buffers(system) {
+                #[cfg(feature = "trace")]
+                let system_span = info_span!("system", name = &*name).entered();
+                self.apply_system_buffers(schedule, world);
+                #[cfg(feature = "trace")]
+                system_span.exit();
+            } else {
+                #[cfg(feature = "trace")]
+                let system_span = info_span!("system", name = &*name).entered();
+                system.run((), world);
+                #[cfg(feature = "trace")]
+                system_span.exit();
+                self.unapplied_systems.set(system_index, true);
```

Comment:
While we're on it, since you're helping maintain `fixedbitset` now, could you add a `remove` function? I still have to do `bitset.set(i, false)` to remove an element and that asymmetry bugs me.

Reply:
Probably a good idea, though I think a new release of `fixedbitset` is probably not in the cards in the immediate future.

Reply:
Issue created upstream.